salt/cloud/clouds/proxmox.py

```diff
@@ -674,17 +675,18 @@ def create(vm_):
     host = data["node"]  # host which we have received
     nodeType = data["technology"]  # VM tech (Qemu / OpenVZ)
 
-    # Determine which IP to use in order of preference:
-    if "ip_address" in vm_:
-        ip_address = str(vm_["ip_address"])
-    elif "public_ips" in data:
-        ip_address = str(data["public_ips"][0])  # first IP
-    elif "private_ips" in data:
-        ip_address = str(data["private_ips"][0])  # first IP
-    else:
-        raise SaltCloudExecutionFailure("Could not determine an IP address to use")
+    if "agent_get_ip" not in vm_ or vm_["agent_get_ip"] == 0:
```

Comment:
This is a setting an user has to set. When this value is set to `1` then Salt-cloud will only try to get the IP address from the agent on the (qemu) VM.

Reply:
And where does the user set it? Is it set in salt-cloud somewhere? Or in the proxmox cloud? My concern is we might be changing default behavior if we are requiring a user to set something they didn't need to before.

Reply:
It should be set in salt-cloud, in the salt-cloud profile configuration.

Reply:
Thanks for clarifying. Can you add documentation for this new setting?

Reply:
Apologies, one more question that I just noticed, I do not see where `agent_get_ip` is getting set to 0 by default.

Reply:
No problem, I'm glad you asked. Actually, it's not set to 0 by default. But setting the value to 0 or not setting the variable at all has the same effect. Should I extend the tests on it? Or should I update the documentation about it? Or should I always set is to 0 to make it simpler? The `if`-statement will then also simpler, because I don't have to test if the variable is set.